source/client/stream_decoder.cc

```diff
@@ -19,6 +19,19 @@ void StreamDecoder::decodeHeaders(Envoy::Http::ResponseHeaderMapPtr&& headers, b
   response_header_sizes_statistic_.addValue(response_headers_->byteSize());
   const uint64_t response_code = Envoy::Http::Utility::getResponseStatus(*response_headers_);
   stream_info_.response_code_ = static_cast<uint32_t>(response_code);
+  const auto timing_header_name = Envoy::Http::LowerCaseString("x-nh-do-not-use-origin-timings");
```

Comment:
Note to reviewers: the odd name here is to scare people away from using this. My plan is to make a new option for nighthawk_client to allow configuration of this header name.

Reply:
re: separately, can you explain why having a configurable header name here is beneficial? just not sure why we wouldn't just dictate this

Well, in my opinion, there's certainly something to say for dictating the names: less code, less configuration. Here are the thoughts I had that made me lean towards allow the end user control of the header names involved. I think this mostly applies to both the client and test-server aspects of it: 

1. Being able to configure the header names both on the server and client side of Nighthawk decouples them. This flexibility may come in handy when mixing Nighthawk components with other OSS or homegrown clients/proxies/test servers with similar capabilities (especially when these dictate the naming involved). 
2. If both the proxy and test server emit timing data using different header names, one can switch which one the clients should track
3. If multiple proxies or test-servers are set up to emit latencies using different names in horizontal scaling, it could be used to track a single instance instead of all of them. 
4. If timing-filter gets extended to emit more timings, and we end up with multiple headers, clients can easily switch which one they track. 
5. In examples the header name could be a little bit longer for clarity, but in final testing, one might want to opt to make it as short as possible to minimise overhead. 

Considering the above, let me know what you think;

Reply:
This makes sense.  Thanks for the explanation.